src/SitemapLinkInterface.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+namespace cebe\luya\sitemap;
+
+/**
+ * Description of SitemapLinkInterface
+ *
+ * @author Rochdi Bazine
+ * @since 1.2.1?
+ */
+interface SitemapLinkInterface
+{
+    /**
+     * Return an iteration of links. If used in ngRestModel, could be useful
+     * to run the find() method
+     * A simple implentation to get all model item would be :
+     * ```php
+     * public function iteratorLinks(){
+     *    return self::find()->all();
+     * }
+     * ```
+     *
+     * @return SitemapLinkInterface.
+     */
+    public function iteratorLinks();
```

Comment:
The problem with `iteratorLinks` is that its not very strict. It would be nice to have an Iterator class which must be returned, but maybe its also just not worth the effort.

To be correct the PHPdoc must be at least: `@return SitemapLinkInterface[]`, and maybe using php 7 array response type hint `public function iteratorLinks() : array;`

But its a good improvement anyhow.

ps: maybe `linksIterator` would be the better method name

Reply:
Okay. Any other suggestions on other methods naming ?

Reply:
Regarding your concern on `iteratorLinks`, I beleive this is somehow handled with the  `getExternalLinkInterfaceObject()` with is verifying that the defined link iterator model is iimplemeting the SitemapLinkInterface interface. But I will add a further check on the retunred objects..